astropy/io/fits/convenience.py

```diff
@@ -497,8 +498,14 @@ def table_to_hdu(table, character_as_bytes=False):
     else:
         table_hdu = BinTableHDU.from_columns(np.array(table.filled()), header=hdr, character_as_bytes=character_as_bytes)
 
-    # Set units for output HDU
+    # Set units and format display for output HDU
     for col in table_hdu.columns:
+
+        if isinstance(col, BaseColumn) and table[col.name].format is not None:
```

Comment:
I would just pick the generic `table[col.name].info.format` so that mixin columns such as `Quantity` can be used as well.
(plus this avoids the `BaseColumn` import...)

Reply:
Here `col` is a fits column, not a Table one, isn't it ?

Reply:
Thanks @mhvk, I didn't realize mixin columns had a format attribute, it was giving me errors when I was calling column.format on the mixins, looks like all unit tests run with the change.